src/IrcConnection/IrcConnection.cs

```diff
@@ -1188,16 +810,27 @@ private class WriteThread
             private int            _BelowMediumThresholdCount = 1;
             private int            _BurstCount;
 
-            public AutoResetEvent QueuedEvent;
+            private AutoResetEvent _QueuedEvent;
 
             /// <summary>
-            /// 
+            /// Fires when a line has been successfully sent to the network stream
+            /// </summary>
+            public event WriteLineEventHandler OnWriteLine;
+
+            /// <summary>
+            /// Set up using the specified connection
             /// </summary>
             /// <param name="connection"></param>
-            public WriteThread(IrcConnection connection)
+            public WriteController(IrcConnection connection)
```

Comment:
WriteThread was a good name before but I moved a lot of WriteLine-type code and the write queues from IrcConnection to WriteThread so it seemed like a good idea to change the class name to avoid confusion.